src/http/client.rs

```diff
@@ -167,49 +161,37 @@ impl<'a> HttpBuilder<'a> {
 
         Ok(self)
     }
-}
 
-impl<'a> Future for HttpBuilder<'a> {
-    type Output = Result<Http>;
+    /// Use the given configuration to build the `Http` client.
+    pub fn build(self) -> Http {
```

Comment:
To prevent `clippy` from generating errors, we need this attribute. The `unwrap`s here are intentional.
```suggestion
    #[allow(clippy::unwrap_used)]
    pub fn build(self) -> Http {
```